.github/workflows/codeql.yml

```diff
@@ -0,0 +1,82 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
```

Comment:
Ideally we should only run C# code for main projects, otherwise, it'll add more time to running all these PR checks.

Reply:
think we can do that since those are test projects. Or should we run this only during our nightly? and PRs

Reply:
I think for test projects we should only run secret scanning and security vulnerability scans, which we already do. CodeQL is not necessary for tests/devapps; I think it'll add more noise - like telling to put comments on public members or use ConfigureAwait and such.

Reply:
>we should only run secret scanning and security vulnerability scans

At the moment, we would like to maintain the same level of analysis for our test projects that we have on our main source project. I think it makes sense so not run it for JS and PY code, but C# should be covered. We are running the same level of Roslyn analysis on our test projects today.

Reply:
Ok. Yea, we can adjust later if needed.